src/util/i18n/i18n.ts

```diff
@@ -95,3 +95,25 @@ export const localizedNumberParser = (
 
   return (formattedString: string): number => pipe(formattedString, transformSeparators, parseFloat)
 }
+
+export type LimitedDateTimeFormatOptions = Pick<
+  Intl.DateTimeFormatOptions,
+  'year' | 'month' | 'day'
+>
+
+export const localizedDateFormatter = (
+  options: LimitedDateTimeFormatOptions,
+  locale: string = 'en-US'
+): Intl.DateTimeFormat => new Intl.DateTimeFormat(locale, options)
+
+export const formatDateToPattern = (
+  patternMap: Record<keyof LimitedDateTimeFormatOptions, string>,
```

Comment:
I'd have named it simply `pattern` or `datePattern` because you suggest here that the input type is a `Map`, or it's a simple record.

Reply:
You are right, I think naming I provided is terrible here.

We can either use `pattern` as you suggest, or define a `Map`, with a more appropriate name, for instance `patternByDateFormatOption`.

What would you suggest?

Thanks

Reply:
I'll suggest to keep it simple. Let's keep the `record` structure and improve the naming for its purpose 😌